cmd/clusterctl/internal/test/fake_objects.go

```diff
@@ -531,10 +532,28 @@ func NewFakeInfrastructureTemplate(name string) *fakeinfrastructure.GenericInfra
 	}
 }
 
+func NewStaticBootstrapConfig() *clusterv1.Bootstrap {
```

Comment:
```suggestion
func NewStaticBootstrapConfig(name string) *clusterv1.Bootstrap {
```

How about doing this and setting the `DataSecretName` to the passed argument. Would be nice to be able to control the value per FakeMachineDeployment instead of using the same for every MachineDeployment.

Also needs a change to `WithStaticBootstrapConfig` to take a name argument.

Reply:
I had that in the beginning. As no test check on the name of the secret itself i decided to use some kind of a static secret.

The only string which is around is the name of the `machinedeployment` itself. We could either use this name as `DataSecretName` or  use it as prefix.

Reply:
IMHO it is better to have unique name rather than use a static name. Let's use the machinedeployment name as a prefix and either way I think you will still have to change `NewStaticBootstrapConfig ` so that it takes a `name string` argument and while calling it from inside `WithStaticBootstrapConfig` use pass `<md-name>-datasecret` as the argument value. 

Thoughts?